addons/mail/static/src/core/persona_service.js

```diff
@@ -55,6 +55,17 @@ export class PersonaService {
             this.store.registeredImStatusPartners?.push(persona.id);
         }
     }
+
+    getMostRecentChatPartners() {
+        const results = [];
+        for (const threadLocalId of Object.values(this.store.discuss.chats.threads)) {
+            const thread = this.store.threads[threadLocalId];
+            if (thread.type === "chat") {
+                results.push(thread.chatPartnerId);
+            }
+        }
+        return results;
+    }
```

Comment:
It's incorrect to assume order of iterating `Object.values(chats)` = `most recent chat partners`.
You should sort by `thread.lastInterestDateTime` instead